src/proxy/momento/src/admin.rs

```diff
@@ -125,55 +125,6 @@ async fn handle_admin_client(mut socket: tokio::net::TcpStream) {
 }
 
 async fn stats_response(socket: &mut tokio::net::TcpStream) -> Result<(), Error> {
-    let mut data = Vec::new();
-    for metric in &metriken::metrics() {
-        let any = match metric.as_any() {
-            Some(any) => any,
-            None => {
-                continue;
-            }
-        };
-
-        // we need to filter some irrelvant metrics that
-        // are defined in crates we depend on
-        if metric.name().starts_with("add")
-            || metric.name().starts_with("append")
-            || metric.name().starts_with("cas")
-            || metric.name().starts_with("decr")
-            || metric.name().starts_with("delete")
-            || metric.name().starts_with("gets")
-            || metric.name().starts_with("incr")
-            || metric.name().starts_with("get_cardinality")
-            || metric.name().starts_with("ping")
-            || metric.name().starts_with("pipeline_depth")
-            || metric.name().starts_with("prepend")
-            || metric.name().starts_with("replace")
-            || metric.name().starts_with("request_latency")
-        {
-            continue;
-        }
-
-        if let Some(counter) = any.downcast_ref::<Counter>() {
-            data.push(format!("STAT {} {}\r\n", metric.name(), counter.value()));
-        } else if let Some(gauge) = any.downcast_ref::<Gauge>() {
-            data.push(format!("STAT {} {}\r\n", metric.name(), gauge.value()));
-        } else if let Some(heatmap) = any.downcast_ref::<Heatmap>() {
-            for (label, value) in PERCENTILES {
-                if let Some(Ok(bucket)) = heatmap.percentile(*value) {
-                    data.push(format!(
-                        "STAT {}_{} {}\r\n",
-                        metric.name(),
-                        label,
-                        bucket.high()
-                    ));
-                }
-            }
-        }
-    }
-
-    data.sort();
-    for line in data {
-        socket.write_all(line.as_bytes()).await?;
-    }
-    socket.write_all(b"END\r\n").await
+    let message = protocol_admin::memcache_stats();
+    socket.write_all(message.as_bytes()).await
```

Comment:
Do we no longer care about filtering the irrelevant metrics or are we taking care of it some other way?

Reply:
This filtering is currently insufficient anyway. A more robust mechanism for disabling stats would be the way to address.